src/axom/primal/geometry/BoundingBox.hpp

```diff
@@ -255,9 +264,9 @@ class BoundingBox
    *  type. This should work as long as the two Ts are comparable with
    *  operator<().
    */
-  template <typename OtherType>
+  template <typename OtherType, int OtherDims>
```

Comment:
Please update the doxygen to reflect that the other box might also have a different dimension

Reply:
Done.

Reply:
Thanks -- actually, this brings up an interesting question about how we should be testing objects of differing dimensions for intersections. 

E.g. when comparing a 3D bounding box against a 2D bounding box the competing approaches would be to:
* consider missing dimensions to have zero extent and to line in the zero plane for that dimension. If so, we should explicitly test for the 3D box covering `z==0` on the 2D box
* ignoring the missing dimensions. This implicitly assumes the missing dimensions infinite extent. I.e. the 2D box ignores the z dimension of the 3D box. This appears to be the current approach.
 
The latter might lead to a lot of false positives if the expectation is that one is testing a 2D computational mesh (that lives in the XY plane) against a 3D tetrahedral "shape". 

I'm wondering if we should remove the new `OtherDims` parameter from `BoundingBox::intersectsWith` and define a more customized intersection routine for bounding boxes with different dimensions in `primal/operators/intersect.hpp`

See: https://github.com/LLNL/axom/blob/620a9c04e987b3849c0d5028885e46430b8e7962/src/axom/primal/operators/intersect.hpp#L398-L408

Reply:
Since there are multiple approaches for how to handle intersections between bounding boxes of different dimensions, I tend to agree that the member function should only take the same dimension. I may just remove that functionality for this merge request until I have a better idea of what my application is doing for this case.

Reply:
I think adding customized intersection routines for cases with bounding boxes with different dims is probably the best way to go for now. This would avoid confusion and potentially undesirable/unexpected results. If common patterns emerge from application use cases, then the interface can be re-examined and tuned for those if it makes sense.

Reply:
For now, I have removed them all together until I know exactly what we need.